File: magellan-library/src/main/java/com/wealthfront/magellan/HistoryRewriter.java
package com.wealthfront.magellan;

import android.app.Activity;

import java.util.Deque;

public interface HistoryRewriter {

  /**
   * Used to rewrite history on the fly, using either {@link Navigator#rewriteHistory(Activity, HistoryRewriter)} or
   * {@link Navigator#navigate(HistoryRewriter)} (and its variants).
   */
  void rewriteHistory(Deque<Screen> history);

}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/HistoryRewriter.kt
package com.wealthfront.magellan.kotlin

import java.util.*

interface HistoryRewriter {

  /**
   * Used to rewrite history on the fly, using either [Navigator.rewriteHistory] or
   * [Navigator.navigate] (and its variants).
   */
  fun rewriteHistory(history: Deque<Screen<*>>)

}